reactor-core/src/main/java/reactor/core/publisher/FluxSwitchOnFirst.java

```diff
@@ -420,64 +419,45 @@ public boolean tryOnNext(T t) {
         }
 
         @Override
-        public void request(long n) {
-            if (Operators.validate(n)) {
-                if (first != null) {
-                    if (drainRegular() && n != Long.MAX_VALUE) {
-                        if (--n > 0) {
-                            s.request(n);
-                            return;
-                        }
-
-                        return;
-                    }
-                }
-
-                s.request(n);
-            }
-        }
-
-        @Override
-        void drain() {
-            drainRegular();
-        }
-
-        boolean drainRegular() {
+        @SuppressWarnings("unchecked")
+        boolean drain() {
             if (WIP.getAndIncrement(this) != 0) {
                 return false;
             }
 
-            T f = first;
+            T f = this.first;
             int m = 1;
             boolean sent = false;
-            @SuppressWarnings("unchecked")
             Fuseable.ConditionalSubscriber<? super T> a =
-                    (Fuseable.ConditionalSubscriber<? super T>) inner;
+                    (Fuseable.ConditionalSubscriber<? super T>) this.inner;
 
             for (;;) {
                 if (f != null) {
-                    first = null;
+                    this.first = null;
 
-                    if (cancelled) {
-                        Operators.onDiscard(f, a.currentContext());
+                    if (a == Operators.emptySubscriber()) {
```

Comment:
I made emptySubscriber an instance of ConditionalSubscriber specifically for that case

Reply:
I'll evaluate that change separately in a comment below (the conditional one). Otherwise, same as elsewhere, can use `EMPTY_SUBSCRIBER` constant.